app/file.go

```diff
@@ -708,7 +708,7 @@ func (a *App) UploadFileX(channelId, name string, input io.Reader,
 
 	if *a.Config().FileSettings.DriverName == "" {
 		return nil, t.newAppError("api.file.upload_file.storage.app_error",
-			"", http.StatusNotImplemented)
+			"api.file.upload_file.storage.app_error", http.StatusNotImplemented)
```

Comment:
This seems to be unintentional. Let's revert this. That parameter is meant to be the error string. In the absence of an error string, an empty field is the correct way.

Reply:
Actually you need to provide something in `details` otherwise you will get this error `always receives "" ...`. In every test case `details` is empty string.

Reply:
Oh I see. Let's remove the `details` parameter then.